piff/basis_interp.py

```diff
@@ -46,51 +46,53 @@ def __init__(self):
         self.degenerate_points = True  # This Interpolator uses chisq quadratic forms
         self.q = None
 
-    def initialize(self, star_list, logger=None):
-        """Initialize the interpolator prefatory to any solve iterations.
-        This class will initialize everything
-        to have constant PSF parameter vector taken from the first Star in the list.
+    def initialize(self, stars, logger=None):
+        """Initialize both the interpolator to some state prefatory to any solve iterations and
+        initialize the stars for use with this interpolator.
 
-        :param star_list:   A list of Star instances to use to initialize.
+        This class will initialize everything to have constant PSF parameter vector taken
+        from the first Star in the list.
+
+        :param stars:       A list of Star instances to use to initialize.
         :param logger:      A logger object for logging debug info. [default: None]
 
         :returns:           A new list of Stars which have their parameters initialized.
         """
+        c = stars[0].fit.params.copy()
+        self.q = c[:,np.newaxis] * self.constant(1.)[np.newaxis,:]
```

Comment:
Right.  This is Gary's code -- I just moved things around a bit (and regularized to using `np.` since we had had a mix of `np` and `numpy`.  

The constant function just makes the appropriate polynomial (at least in BasisPolynomial it is a polynomial) that corresponds to a constant value in (u,v).  I could add a base class method that raises NotImplemented with an appropriate message saying that you need to use a concrete derived class, not the abstract base class.